Update docs to refer to init and newer aurora (#364)
kairos-io/kairos-docs: content/en/docs/Reference/kairos-factory.md

| -l           | Sets the log level                                       | info,warn,debug,trace       | info            |
| -s           | Sets the stage to run                                    | install, init, all          | all             |
| -m           | Sets the model                                           | generic, rpi3, rpi4         | generic         |
| -v           | Sets the variant                                         | core, standard              | core            |
| -k           | Sets the Kubernetes provider                             | k3s,k0s                     | k3s             |
| -r           | Sets registry and org (Checks this repo when upgrading)  | Any valid registry and org  | quay.io/kairos  |
| -t           | Sets Trusted Boot on                                     | true,false                  | false           |
| -f           | Sets the framework version to use                        | Any valid framework version | latest          |
| --fips       | Use FIPS framework for FIPS 140-2 compliance images      | true,false                  | false           |

> | --k8sversion | Sets the provider version to use                         | Any valid provider version  | latest          |

Reviewer: can't we already say: k3s or k0s ?

Author: well, those are providers no? So any valid provider version is correct and it opens to have more providers (rke2 maybe?) so I think this is correct and concise.

If you want to override those see the flags `-k` and `--k8sversion`

{{% /alert %}}

## Phases

kairos-init is divided in 2 phases, one its the install phase which install all nedeed packages and binaries and the other is the init phase, which gets the system ready. This are the main parts of each phase:

Install:
 - Install required packages via system package manager
 - Install [kairos-framework](https://github.com/kairos-io/kairos-framework) into the system
 - Install the Kairos' provider, including a k8s distribution and tools if it's a standard image

Init:
 - Fill the /etc/kairos-release data (needed for upgrades, grub booting, kernel cmdline, etc...)
 - Remove Old kernel links and duplicates
 - Get and link kernel to `/boot/vmlinuz`
 - Remove all existing initrds
 - Build a new initrd and link it to `/boot/initrd`
 - Enable/disable needed services
 - Run some workarounds (grub vs grub2 naming for example)
 - Run system cleanup to avoid leftovers

> As you can see, both of these stages runs separatedly so you can hijack this in the middle and add or remove things.

Reviewer: ```suggestion
As you can see, both of these stages runs separately so you can hijack this in the middle and add or remove things.
```

```Dockerfile
ARG BASE_IMAGE=ubuntu:20.04

FROM quay.io/kairos/kairos-init:v0.2.6 AS kairos-init

FROM ${BASE_IMAGE} AS base-kairos
ARG VARIANT=core
ARG MODEL=generic
ARG TRUSTED_BOOT=false
ARG KUBERNETES_DISTRO=k3s
ARG KUBERNETES_VERSION=latest
ARG FRAMEWORK_VERSION=v2.16.1
ARG VERSION

COPY --from=kairos-init /kairos-init /kairos-init
RUN /kairos-init -f "${FRAMEWORK_VERSION}" -l debug -s install -m "${MODEL}" -v "${VARIANT}" -t "${TRUSTED_BOOT}" -k "${KUBERNETES_DISTRO}" --k8sversion "${KUBERNETES_VERSION}" --version "${VERSION}"
RUN /kairos-init -f "${FRAMEWORK_VERSION}" -l debug -s init -m "${MODEL}" -v "${VARIANT}" -t "${TRUSTED_BOOT}" -k "${KUBERNETES_DISTRO}" --k8sversion "${KUBERNETES_VERSION}" --version "${VERSION}"
RUN /kairos-init -f "${FRAMEWORK_VERSION}" -l debug --validate -m "${MODEL}" -v "${VARIANT}" -t "${TRUSTED_BOOT}" -k "${KUBERNETES_DISTRO}" --k8sversion "${KUBERNETES_VERSION}" --version "${VERSION}"
RUN rm /kairos-init
```

### Build rpi4 k3s oci artifacts (arm64 platform)

Based on Alpine 3.19:

```bash

> $ docker build --platform=arm64 -t alpine-rpi:3.19 --build-arg MODEL=rpi4 --build-arg BASE_IMAGE=alpine:3.19 --build-arg VERSION=v1.0.0 .

Reviewer: does Alpine work? :p

Author: it will.

RUN /kairos-init -l debug -s install -m "${MODEL}" -v "${VARIANT}" -t "${TRUSTED_BOOT}" -k "${KUBERNETES_DISTRO}" --k8sversion "${KUBERNETES_VERSION}" --version "${VERSION}"
RUN /kairos-init -l debug -s init -m "${MODEL}" -v "${VARIANT}" -t "${TRUSTED_BOOT}" -k "${KUBERNETES_DISTRO}" --k8sversion "${KUBERNETES_VERSION}" --version "${VERSION}"
RUN /kairos-init -l debug --validate -m "${MODEL}" -v "${VARIANT}" -t "${TRUSTED_BOOT}" -k "${KUBERNETES_DISTRO}" --k8sversion "${KUBERNETES_VERSION}" --version "${VERSION}"
RUN rm /kairos-init
```

{{% alert title="Variant" color="note" %}}

When selecting the `standard` variant, the produced image will contain the default kubernetes provider (k3s) with the latest published version of k3s, the Kairos provider for kubernetes and some extra k8s utils like k9s or kube-vip.

If you want to override those see the flags `-k` and `--k8sversion`

{{% /alert %}}

## Phases

kairos-init is divided in 2 phases, one its the install phase which install all nedeed packages and binaries and the other is the init phase, which gets the system ready. This are the main parts of each phase:

Install:
 - Install required packages via system package manager

>  - Install [kairos-framework]() into the system

Reviewer 2: ```suggestion
 - Install [kairos-framework](https://github.com/kairos-io/kairos-framework) into the system
```

 - Install k8s provider and tools if its an standard image

Init:
 - Fill the /etc/kairos-release data (needed for upgrades, grub booting, kernel cmdline, etc...)
 - Remove Old kernel links and duplicates
 - Get and link kernel to `/boot/vmlinuz`
 - Remove all existing initrds
 - Build a new initrd and link it to `/boot/initrd`
 - Enable/disable needed services
 - Run some workarounds (grub vs grub2 naming for example)
 - Run system cleanup to avoid leftovers

As you can see, both of this stages runs separatedly so you can hijack this in the middle and add or remove things.

> For example, its possible to add extra modules to be added to the initrd, or a specific kernel instead of the default latest one.

Reviewer 2: ```suggestion
For example, it's possible to add extra modules to be added to the initrd, or a specific kernel instead of the default latest one.
```

Extra services can be either added or made required, etc..

> The separation its also very useful for caching, as once the install phase has been cached by docker we can modify the following steps to fix any issues before the init phase is run, without removing the cache, which is really good under cross platforms like aarch64 where installing packages is very slow unless you are on the native platform. 

Reviewer 2: ```suggestion
The separation it's also very useful for caching, as once the install phase has been cached by docker we can modify the following steps to fix any issues before the init phase is run, without removing the cache. This is particularly useful under cross platform builds where speed can take a big impact depending on your setup. 
```
